spec/unit/face/parser_spec.rb

```diff
@@ -98,6 +98,16 @@
       expect(@logs[0].message).to eq("Syntax error at end of file")
       expect(@logs[0].level).to eq(:err)
     end
+
+    it "logs an error if the input begins with a UTF-8 BOM" do
+      utf8_bom_manifest = file_containing('utf8_bom.pp', "\uFEFFnotice hi")
+
+      output = parser.dump(utf8_bom_manifest)
+
+      expect(output).to eq("")
+      expect(@logs[1].message).to eq("Syntax error at '' at line 1:1")
```

Comment:
we should be able to do better than that - the lexer could raise a lexing error instead when it sees FEFF or FFFE as the first two bytes.